app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt

```diff
@@ -159,4 +159,21 @@ class JupiterSwapInteractor(
             is SwapState.SwapException -> null
         }
     }
+
+    /**
+     * We need to cast BigIntegers to Double to keep numbers after dot when dividing.
+     * We had an error when we divided 2039280 / 955187 and got only 2 instead of full 2.13
+     * which caused invalid behaviour in case of slippage 2.1%
+     */
+    private fun checkForHighFees(
+        totalFeesAndDeposit: BigInteger,
+        outAmount: BigInteger,
+        slippageValuePercent: Int
+    ): Boolean {
+        val feesValuePercent: Int =
+            (totalFeesAndDeposit.toDouble() / outAmount.toDouble())
```

Comment:
why not just use `BigInteger.setScale(precision)`, or even `totalFeesAndDeposit.divide(outAmount, precision, RoundingMode. HALF_EVEN)`